docs/flatcar-linux/kvm-libvirt.md

```diff
@@ -0,0 +1,349 @@
+but for example in Ubuntu you would use `apt` as stated in the comment in the same line.
+
+```sh
+$ sudo dnf install libvirt-daemon  # or sudo apt install libvirt-daemon
+$ sudo systemctl start libvirtd
+```
+
+## User Setup
+
+Ensure that libvirt is accessible for the current user:
+
+```sh
+$ sudo usermod -a -G libvirt $(whoami)
+$ newgrp libvirt
+```
+
+## Terraform Setup
+
+Install [Terraform](https://www.terraform.io/downloads.html) v0.11.x on your system.
+
+```sh
+$ terraform version
+Terraform v0.11.13
+```
+
+Add the [terraform-provider-ct](https://github.com/poseidon/terraform-provider-ct) plugin binary for your system
+to `~/.terraform.d/plugins/`, noting the `_v0.3.1` suffix.
+
+```sh
+wget https://github.com/poseidon/terraform-provider-ct/releases/download/v0.3.1/terraform-provider-ct-v0.3.1-linux-amd64.tar.gz
+tar xzf terraform-provider-ct-v0.3.1-linux-amd64.tar.gz
+mv terraform-provider-ct-v0.3.1-linux-amd64/terraform-provider-ct ~/.terraform.d/plugins/terraform-provider-ct_v0.3.1
+```
+
+Add the [terraform-provider-libvirt](https://github.com/dmacvicar/terraform-provider-libvirt) plugin binary for your system
+to `~/.terraform.d/plugins/`, noting the `_v0.5.2` suffix.
+You should use the 0.5.2 release because it includes the experimental
+[pool definition feature](https://github.com/dmacvicar/terraform-provider-libvirt/commit/9f00f3d46c489f24c71d02fde816f5fda34d3f7c)
+but building from source is also an option.
+
+```sh
+wget https://github.com/dmacvicar/terraform-provider-libvirt/releases/download/v0.5.2/terraform-provider-libvirt-0.5.2.Fedora_28.x86_64.tar.gz
+tar xzf terraform-provider-libvirt-0.5.2.Fedora_28.x86_64.tar.gz
+mv terraform-provider-libvirt ~/.terraform.d/plugins/terraform-provider-libvirt_v0.5.2
+```
+
+
+Read [concepts](/docs/architecture/concepts.md) to learn about Terraform, modules, and organizing resources if the following confuses you.
+
+A Terrafrom project should be created in a new clean directory.
+Assuming there is a parent directory `infra` for all the terraform projects, we create a new subdirectory `vmcluster`.
+
+```sh
+mkdir vmcluster
+cd vmcluster
+```
+
+## Provider
+
+Create a file named `provider.tf` with the following contents:
+
+```tf
+provider "ct" {
+  version = "0.3.1"
+}
+
+provider "local" {
+  version = "~> 1.0"
+  alias   = "default"
+}
+
+provider "null" {
+  version = "~> 1.0"
+  alias   = "default"
+}
+
+provider "template" {
+  version = "~> 1.0"
+  alias   = "default"
+}
+
+provider "tls" {
+  version = "~> 1.0"
+  alias   = "default"
+}
+
+provider "libvirt" {
+  version = "~> 0.5.2"
+  uri = "qemu:///system"
+  alias = "default"
+}
+```
+
+## Cluster
+
+Define a Kubernetes cluster using the module
+[kvm-libvirt/flatcar-linux/kubernetes](https://github.com/kinvolk/lokomotive-kubernetes/tree/master/kvm-libvirt/flatcar-linux/kubernetes).
+
+Create a file called `mycluster.tf` with these contents but remember to change the `CHANGEME` values:
```

Comment:
Rather than relying on user reading docs, can we make CHANGEME values a required variables here?